js/main.js

```diff
@@ -20,4 +20,77 @@ $entryForm.addEventListener('submit', function (event) {
   data.entries.unshift(submittedData);
   $image.setAttribute('src', 'images/placeholder-image-square.jpg');
   $entryForm.reset();
+
+  const newData = renderEntry(submittedData);
+  $unorderedList.prepend(newData);
+  toggleNoEntries();
+  viewSwap('entries');
 });
+
+function renderEntry(entry) {
+  const $newLi = document.createElement('li');
+  const $rowDiv = document.createElement('div');
+  const $columnHalfDiv = document.createElement('div');
+  const $imgElement = document.createElement('img');
+  const $secondColumnHalf = document.createElement('div');
+  const $titleElement = document.createElement('h3');
+  const $noteElement = document.createElement('p');
+
+  $rowDiv.setAttribute('class', 'row');
+  $columnHalfDiv.setAttribute('class', 'column-half');
+  $imgElement.setAttribute('class', 'image');
+  $imgElement.setAttribute('src', entry.url);
+  $secondColumnHalf.setAttribute('class', 'column-half');
+  $titleElement.textContent = entry.title;
+  $noteElement.textContent = entry.notes;
+
+  $newLi.appendChild($rowDiv);
+  $rowDiv.appendChild($columnHalfDiv);
+  $columnHalfDiv.appendChild($imgElement);
+
+  $rowDiv.appendChild($secondColumnHalf);
+  $secondColumnHalf.appendChild($titleElement);
+  $secondColumnHalf.appendChild($noteElement);
+
+  return $newLi;
+}
+
+const $unorderedList = document.querySelector('.no-bullets');
+
+document.addEventListener('DOMContentLoaded', function (event) {
+  for (let i = 0; i < data.entries.length; i++) {
+    $unorderedList.appendChild(renderEntry(data.entries[i]));
+  }
+  viewSwap(data.view);
+  toggleNoEntries();
+});
+
+function toggleNoEntries() {
+  const $noEntryText = document.querySelector('.no-entry');
```

Comment:
`querySelector`s should always try to be done globally. By doing them inside of a function, you're forcing us to traverse the DOM tree everytime this function gets called to find this element. By moving it outside of any functions, you only do the search once, and we will always have a reference to it